integration/basic_test.go

```diff
@@ -604,6 +604,53 @@ func TestDebugError(t *testing.T) {
 	})
 }
 
+func TestCheckingNestedDocuments(t *testing.T) {
+	t.Parallel()
+
+	for name, tc := range map[string]struct {
+		doc any
+		err error
+	}{
+		"1ok": {
+			doc: CreateNestedDocument(1),
+		},
+		"10ok": {
+			doc: CreateNestedDocument(10),
+		},
+		"100ok": {
+			doc: CreateNestedDocument(100),
+		},
+		"179ok": {
+			doc: CreateNestedDocument(179),
+		},
```

Comment:
Currently we only test nesting limit for document, could you test also array as well? Is something like `[{doc:[{doc:[...]}]}]`  not possible? :thinking:

Reply:
The only type that can be inserted is the document, any other type like array is inside provided document, right?

Reply:
FWIW, I think it is not a problem to support arrays inside arrays there, even if they are rejected at a later layer (just like duplicate fields, for example)

Reply:
Yeah that's right. Let's add a simple test case for nested array then :+1:

Reply:
How do we want to test nested arrays in integration tests? They'll behave differently on Mongodb

Reply:
By not testing it with integration tests, I guess :)

Reply:
Well, to create an unit test we might need to provide binary representation as an input, and I'm not sure if that's what we do

Reply:
Maybe I'm misunderstanding how nesting validation works. But if we have document nested in array nested in document etc, only 179 level of nesting is valid right? This is allowed by both so I feel that is testable by integration test.

As for pure nested array, I _thought_ that is validated after we check for 179 level of nesting (maybe I'm wrong). If that's that case, we can test nesting array of 197 and check that it returns error in integration test?

Reply:
Actually testing array is already done and I didn't understand `createNestedDocument` function :see_no_evil: